src/datachain/lib/dc.py

```diff
@@ -1513,22 +1514,41 @@ def from_records(
         Parameters:
             to_insert : records (or a single record) to insert. Each record is
                         a dictionary of signals and theirs values.
+            schema : describes chain signals and their corresponding types
 
         Example:
             ```py
-            empty = DataChain.from_records()
             single_record = DataChain.from_records(DataChain.DEFAULT_FILE_RECORD)
             ```
         """
         session = Session.get(session)
         catalog = session.catalog
 
+        if not to_insert and not schema:
+            raise ValueError("Non empty records to insert or schema must be defined")
```

Comment:
can we define both? what takes precedence then?

Reply:
btw, why can't we do a completely empty schema and no records?

Reply:
Yea, I removed this exception. Now if empty records and no schema is set, default columns are created (as in general when no schema is set) ... in followup issue we should try to infer schema from records, if explicit schema is not defined, and then if there is no records no columns will be created

Reply:
About precedence, schema now takes precedence if it's defined

Reply:
> default columns are created 

what are the default columns though? It probably comes back to my question re the `DEFAULT_FILE_RECORD` - I don't understand its meaning tbh.